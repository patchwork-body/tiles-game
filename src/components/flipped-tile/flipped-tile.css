.flipped-tile {
  border-radius: inherit;
  min-width: 100%;
  min-height: 100%;
}

.flipped-tile_hidden {
  opacity: 0;
}
