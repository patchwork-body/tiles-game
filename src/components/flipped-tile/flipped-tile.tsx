import React, { SyntheticEvent, useCallback } from 'react';

const FlippedTile: FlippedTileComponent = ({
  index,
  color,
  hidden,
  clickHandler,
}) => {
  const classList = ['flipped-tile'];

  if (hidden) {
    classList.push('flipped-tile_hidden');
  }

  const onClick = useCallback(
    (event: SyntheticEvent) => {
      const index = Number(event.currentTarget.getAttribute('data-index'));
      clickHandler(index);
    },
    [clickHandler],
  );

  return (
    <div
      data-index={index}
      className={classList.join(' ')}
      style={{ background: color }}
      onClick={onClick}
    ></div>
  );
};

export { FlippedTile };
