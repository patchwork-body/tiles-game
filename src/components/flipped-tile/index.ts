import './flipped-tile.css';

export * from './flipped-tile';
