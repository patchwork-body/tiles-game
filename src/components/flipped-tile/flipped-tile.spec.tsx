import React from 'react';
import { render } from '@testing-library/react';

import { FlippedTile } from './flipped-tile';

describe('GridTile', () => {
  const index = 0;
  const color = 'black';
  const hidden = true;
  const clickHandler = jest.fn();

  const props: FlippedTileProps = {
    index,
    color,
    hidden,
    clickHandler,
  };

  test('should be visible', () => {
    const { container } = render(
      <FlippedTile {...{ ...props, hidden: false }} />,
    );
    expect(container.classList.contains('tile_hidden')).toBeFalsy();
  });

  test('should match snapshot', () => {
    const { container } = render(<FlippedTile {...props} />);
    expect(container.firstElementChild).toMatchSnapshot();
  });
});
