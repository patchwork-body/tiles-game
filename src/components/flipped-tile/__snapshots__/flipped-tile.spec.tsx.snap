// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridTile should match snapshot 1`] = `
<div
  class="flipped-tile flipped-tile_hidden"
  data-index="0"
  style="background: black;"
/>
`;
